MLS.Agent.Tests/CommandLine/PublishCommandTests.cs

```diff
@@ -36,20 +49,69 @@ public class WithMarkdownOutputFormat
             [InlineData("[link](https://try.dot.net/)")]
             public async Task When_there_are_no_code_fence_annotations_markdown_is_unchanged(string markdown)
             {
-                var rootDirectory = Create.EmptyWorkspace(isRebuildablePackage: true).Directory;
-
-                var directoryAccessor = new InMemoryDirectoryAccessor(rootDirectory)
-                {
+                var files = PrepareFiles(
                     ("doc.md", markdown)
-                }.CreateFiles();
+                );
 
+                var (publishOutput, resultCode) = await DoPublish(files).ConfigureAwait(false);
 
-                var publishOutput = await DoPublish(directoryAccessor).ConfigureAwait(false);
-
+                resultCode.Should().Be(0);
                 publishOutput.OutputFiles.Single().Content.Should().Be(markdown);
             }
 
-            private async Task<PublishOutput> DoPublish(IDirectoryAccessor source)
+            [Theory]
+            [InlineData(@"
+## C# null coalesce example
+``` cs --source-file ./project/Program.cs --region null_coalesce --project ./project/some.csproj
+```
+")]         [InlineData(@"
+## C# null coalesce example
+``` cs --source-file ./project/Program.cs --region null_coalesce --project ./project/some.csproj
+    var length = some buggy c# example code
+    
+```
+")]
+            public async Task When_there_are_code_fence_annotations_in_markdown_content_of_the_fenced_section_is_replaced_with_the_project_code(string markdown)
+            {
+                var files = PrepareFiles(
+                    ("./folder/project/some.csproj", CsprojContents),
+                    ("./folder/project/Program.cs", CompilingProgramWithRegionCs),
+                    ("./folder/doc.md", markdown)
+                );
+
+                var (publishOutput, resultCode) = await DoPublish(files).ConfigureAwait(false);
```

Comment:
You shouldn't need `ConfigureAwait` in the tests. We're not doing this elsewhere. If we're seeing a deadlock, there's a bug somewhere in our product code and we can track it down.